data/src/test/java/org/oppia/android/data/backends/gae/RemoteAuthNetworkInterceptorTest.kt

```diff
@@ -0,0 +1,170 @@
+package org.oppia.android.data.backends.gae
+
+import android.app.Application
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.core.content.pm.ApplicationInfoBuilder
+import androidx.test.core.content.pm.PackageInfoBuilder
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import okhttp3.Headers
+import okhttp3.OkHttpClient
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.data.backends.gae.api.TopicService
+import org.oppia.android.testing.network.MockTopicService
+import org.robolectric.Shadows
+import org.robolectric.annotation.LooperMode
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import retrofit2.mock.MockRetrofit
+import retrofit2.mock.NetworkBehavior
+import javax.inject.Inject
+import javax.inject.Qualifier
+import javax.inject.Singleton
+
+/** Tests for [RemoteAuthNetworkInterceptor] */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+class RemoteAuthNetworkInterceptorTest {
+
+  @Inject lateinit var networkInterceptor: RemoteAuthNetworkInterceptor
+
+  @Inject lateinit var context: Context
+
+  private lateinit var mockRetrofit: MockRetrofit
+
+  private lateinit var retrofit: Retrofit
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    setUpApplicationForContext()
+    setUpMockRetrofit()
+  }
+
+  private fun setUpApplicationForContext() {
+    val packageManager = Shadows.shadowOf(context.packageManager)
+    val applicationInfo =
+      ApplicationInfoBuilder.newBuilder()
+        .setPackageName("test_package_name")
+        .build()
+    val packageInfo =
+      PackageInfoBuilder.newBuilder()
+        .setPackageName("test_package_name")
+        .setApplicationInfo(applicationInfo)
+        .build()
+    packageInfo.versionName = "1.0"
+    packageInfo.versionCode = 1
+    packageManager.installPackage(packageInfo)
```

Comment:
This doesn't seem to set the versionName or versionCode -- it throws a NullPointerException later on in the test for the versionName when the RemoteAuthNetworkInterceptor tries to set the headers ([line 44)](https://github.com/oppia/oppia-android/pull/2907/files#diff-c443970e72f2a145c0321816f33ec349b2cdf2054e8bc3b2f0338cb742ef2cf2R44). Any tips?

Reply:
I think NPE can happen in Context.getVersionName() if there's no PackageInfo for the current package name. You should be able to debug in this test what the package name is for Context and whether it's matching the PackageInfo you're creating (which is using package name 'test_package_name').